build_runner/lib/src/generate/create_merged_dir.dart

```diff
@@ -22,11 +22,28 @@ final _logger = new Logger('CreateOutputDir');
 const _manifestName = '.build.manifest';
 const _manifestSeparator = '\n';
 
-/// Creates a merged output directory for a build at [outputPath].
+/// Creates merged output directories for each value in [outputMap].
 ///
 /// Returns whether it succeeded or not.
-Future<bool> createMergedOutputDir(
+Future<bool> createMergedOutputDirectories(
+    Map<String, String> outputMap,
+    AssetGraph assetGraph,
+    PackageGraph packageGraph,
+    AssetReader reader,
+    BuildEnvironment environment,
+    List<BuildPhase> buildPhases) async {
+  for (var output in outputMap.keys) {
+    if (!await _createMergedOutputDir(output, outputMap[output], assetGraph,
```

Comment:
should we parallelize these at all?

```dart
...) {
  _create(entry) => _createMergedOutputDir(entry.key, entry.value, assetGraph,
      packageGraph, reader, environment, buildPhases);
  return Future
      .wait(outputMap.entries.map(_create))
      .then((r) => r.every((r) => r));
}
```

Reply:
Discussed with @jakemac53 already and since this will be limited by the number of file handlers we likely won't get much of a performance boost. Leaving as is. We can revisit later. We'll have to be more clever with the logging.